File: ProyectoSpring/SpringBussinesOne/demo/demo/src/main/java/com/BussinesOne/demo/models/Dtos/PerfilDTO.java
package com.BussinesOne.demo.models.Dtos;

import java.time.LocalDate;
import java.util.List;

import lombok.Data;

@Data
public class PerfilDTO {
    private Long id;
    private String nombre;
    private String email;
    private String rol;
    private String direccion;
    private String infoPersonal;
    private String formacionAcademica;
    private String datosPersonales;

    private String contrasena;
    private String telefono;
    private String dni;
    private LocalDate fechaNacimiento;
    private String numeroSeguridadSocial;
    private String iban;

    private List<ProductoSinUsuarioDTO> productos;

    @Data
    public static class ProductoSinUsuarioDTO {
    private Long codigo;
    private String nombre;
    private double precio;
    private String foto;
    }
}

File: ProyectoAndroid/app/src/main/java/com/example/proyectobussinesone/navigation/models/PerfilDto.kt
package com.example.proyectobussinesone.navigation.models

import com.example.proyectobussinesone.ui.viewmodel.Perfil
import com.google.gson.annotations.SerializedName

data class PerfilDto(
    @SerializedName("id") val id: Long,
    @SerializedName("nombre") val nombre: String,
    @SerializedName("contrasena") val contraseña: String?,
    @SerializedName("email") val email: String,
    @SerializedName("telefono") val telefono: String?,
    @SerializedName("direccion") val direccion: String?,
    @SerializedName("dni") val dni: String?,
    @SerializedName("fechaNacimiento") val fechaNacimiento: String?,
    @SerializedName("numeroSeguridadSocial") val numeroSeguridadSocial: String?,
    @SerializedName("iban") val iban: String?,
    @SerializedName("formacionAcademica") val formacionAcademica: String?, // si lo devuelves como lista
    @SerializedName("datosPersonales") val datosPersonales: String?
)

fun PerfilDto.toPerfil(): Perfil {
    return Perfil(
        id = this.id,
        nombre = this.nombre,
        email = this.email,
        contraseña = this.contraseña,
        telefono = this.telefono,
        direccion = this.direccion,
        formacionAcademica = this.formacionAcademica,
        datosPersonales = this.datosPersonales,
        dni = this.dni,
        fechaNacimiento = this.fechaNacimiento,
        numeroSeguridadSocial = this.numeroSeguridadSocial,
        iban = this.iban
    )
}
